packages/radio-player/demo/radio-player-controller.ts

```diff
@@ -48,15 +40,54 @@ export default class RadioPlayerController extends LitElement {
     `;
   }
 
-  async loadItemMetadata() {
+  /* eslint-disable @typescript-eslint/no-explicit-any */
+  async loadItemMetadata(): Promise<any> {
     const url = `https://archive.org/metadata/${this.itemId}`;
     const response = await fetch(url);
     const data = await response.json();
 
     this.handleMetadataResponse(data);
   }
 
-  handleMetadataResponse(response: any) {
+  async fetchTranscript(): Promise<any> {
+    const srtUrl = `https://archive.org/cors/${this.itemId}/${this.fileName}`;
+
+    const response = await fetch(srtUrl);
+    const json = await response.json();
+
+    const transcriptEntries = json.map(
+      (entry: any) =>
+        new TranscriptEntryConfig(
+          entry.id,
+          entry.start,
+          entry.end,
+          entry.text,
+          entry.is_music,
+          entry.search_match_index,
+        ),
+    );
+
+    this.transcriptConfig = new TranscriptConfig(transcriptEntries);
+  }
+
+  async setup(): Promise<any> {
+    await this.loadItemMetadata();
+    await this.fetchTranscript();
+
+    const searchParams = new URLSearchParams(window.location.search);
+    const searchTerm = searchParams.get('q');
+    const startTime = searchParams.get('start');
+
+    if (searchTerm && this.radioPlayer) {
+      this.radioPlayer.searchTerm = searchTerm;
```

Comment:
In practice, no, but this is TypeScript being strict about the possibility that it might not. `this.radioPlayer` is defined this way:
```
  private get radioPlayer(): RadioPlayer | null {
    return this.shadowRoot ? (this.shadowRoot.querySelector('radio-player') as RadioPlayer) : null;
  }
```
Since it's possible for there to not be a `shadowRoot`, as well as the possibility for that `shadowRoot.querySelector()` to return nothing, you have to assume that it's possible it won't exist.

In TypeScript 3.7, they added this super nice feature called Optional Chaining that would've let me do this:
```
private get radioPlayer(): RadioPlayer | null {
  return this.shadowRoot?.querySelector('radio-player') as RadioPlayer;
}
...
if (searchTerm) {
  this.radioPlayer?.searchTerm = searchTerm;
}
```